code/controllers/subsystem/profiler.dm

```diff
@@ -37,14 +41,34 @@ SUBSYSTEM_DEF(profiler)
 // Write the file while also cost tracking
 /datum/controller/subsystem/profiler/proc/DumpFile()
 	var/timer = TICK_USAGE_REAL
+	// Fetch info
 	var/current_profile_data = world.Profile(PROFILE_REFRESH, format = "json")
 	fetch_cost = MC_AVERAGE(fetch_cost, TICK_DELTA_TO_MS(TICK_USAGE_REAL - timer))
 	CHECK_TICK
 	if(!length(current_profile_data)) //Would be nice to have explicit proc to check this
 		stack_trace("Warning, profiling stopped manually before dump.")
 	var/json_file = file("[GLOB.log_directory]/profile.json")
+	// Put it in a file
 	if(fexists(json_file))
 		fdel(json_file)
 	timer = TICK_USAGE_REAL
 	WRITE_FILE(json_file, current_profile_data)
 	write_cost = MC_AVERAGE(write_cost, TICK_DELTA_TO_MS(TICK_USAGE_REAL - timer))
+
+	// Send it down redis
+	if(SSredis.connected)
+		// Encode
+		timer = TICK_USAGE_REAL
+
+		var/list/ffi_data = list()
+		ffi_data["round_id"] = GLOB.round_id
+		// We dont have to JSON decode here. The other end can worry about a 2-layer decode.
+		// Performance matters on this end. It doesnt on the other end
+		ffi_data["profile_data"] = current_profile_data
```

Comment:
Going to see how expensive encode/decode is here, as it may decrease send time due to the send size being much smaller from not having tons of extra escapes in
![image](https://user-images.githubusercontent.com/25063394/155905163-71a984c4-c11d-427a-955e-e0954ffe586c.png)

Reply:
Update:

lol no
![image](https://user-images.githubusercontent.com/25063394/155905297-7d987b89-d7f4-4bdf-9386-a5d55433edb8.png)